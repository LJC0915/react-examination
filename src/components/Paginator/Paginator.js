import React from "react";
import styles from "./Paginator.scss";
import PropTypes from "prop-types";
import { range } from "lodash";

export const Paginator = ({ currentPage, totalPage, changePage }) => {
  const pageButtons = range(1, totalPage + 1);
  const onClickPage = page => {
    changePage(page);
  };
  return (
    <div className={styles.wrap}>
      <div>
        <button
          className={`btn btn-default ${styles.btnAction}`}
          disabled={currentPage <= 1}
          onClick={() => onClickPage(currentPage - 1)}
        >
          Prev
        </button>
      </div>
      <div>
        {pageButtons.map((page, index) => (
          <button
            key={index}
            className={`btn ${
              currentPage === page ? "btn-primary" : "btn-default"
            } ${styles.btnAction}`}
            onClick={() => onClickPage(page)}
          >
            {page}
          </button>
        ))}
      </div>
      <div>
        <button
          className={`btn btn-default ${styles.btnAction}`}
          disabled={currentPage === totalPage}
          onClick={() => onClickPage(currentPage + 1)}
        >
          Next
        </button>
      </div>
    </div>
  );
};

Paginator.propTypes = {
  changePage: PropTypes.func.isRequired,
  currentPage: PropTypes.number.isRequired,
  totalPage: PropTypes.number.isRequired
};
