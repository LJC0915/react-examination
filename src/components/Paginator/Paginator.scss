:local(.wrap) {
  display: flex;
  justify-content: space-between;
  background: #fff;
  padding: 10px 0;
  width: 100%;
}

:local(.btnAction),
:local(.btnAction):active,
:local(.btnAction):focus,
:local(.btnAction):hover {
  margin:0 5px;
}
