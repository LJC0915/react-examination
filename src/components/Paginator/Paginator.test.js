import React from "react";
import renderer from "react-test-renderer";
import { Paginator } from "./Paginator";

describe("Paginator spec", () => {
  it("render Paginator correctly", () => {
    const mockChangePage = jest.fn();
    const wrapper = renderer
      .create(
        <Paginator totalPage={2} currentPage={1} changePage={mockChangePage} />
      )
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
