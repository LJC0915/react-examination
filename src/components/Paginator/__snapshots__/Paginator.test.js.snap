// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator spec render Paginator correctly 1`] = `
<div>
  <div>
    <button
      className="btn btn-default undefined"
      disabled={true}
      onClick={[Function]}
    >
      Prev
    </button>
  </div>
  <div>
    <button
      className="btn btn-primary undefined"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="btn btn-default undefined"
      onClick={[Function]}
    >
      2
    </button>
  </div>
  <div>
    <button
      className="btn btn-default undefined"
      disabled={false}
      onClick={[Function]}
    >
      Next
    </button>
  </div>
</div>
`;
